front/Dockerfile

```diff
@@ -1,5 +1,6 @@
 FROM node@sha256:1a9a71ea86aad332aa7740316d4111ee1bd4e890df47d3b5eff3e5bded3b3d10 AS build-deps
 
+RUN apk add --no-cache python3 make g++
```

Comment:
Why is this required ? Is this arm64 only ?
What requires Python here ?

I could track this down to https://github.com/anthares101/CTFNote/commit/2aff33507ff52816a44e114271d54480c133eec1 but I am not sure what is going on.
Is this related to a node update ? If yes, does this apply to us ?

Reply:
It is quite a long time ago, but I remember that this was related to node-gyp on arm64 builds of CTFNote. These tools should be installed in order to build CTFNote successfully.

Reply:
Ah, thanks.
Now that I think about it, I'm surprised this works though : the images are pinned to amd64 versions. How does that even work ? Is Docker flat out ignoring digests ?

Reply:
I do not understand your question unfortunately. I think these build tools are probably required by `yarn` in order to compile certain native dependencies into another architecture. Therefore I think it worked without arm64 but now these are required since we include arm64.

Reply:
Ah, sorry for the confusion.

My question refers to this line in particular :
```
FROM node@sha256:1a9a71ea86aad332aa7740316d4111ee1bd4e890df47d3b5eff3e5bded3b3d10
```

This container is pinned to a specific digest.

If we try to use a digest that is not for our architecture, Podman warns us about that :
```
Trying to pull docker.io/library/node@sha256:bd59aa24c8e6d84170b0f4f17e8b94ba21ee6a70b989e81bb793af93dad351a3...
[...]
WARNING: image platform (linux/s390x) does not match the expected platform (linux/amd64)
```

And when I try to create a container from the newly-pulled image, both Docker and Podman are panicking :
```
WARNING: The requested image's platform (linux/s390x) does not match the detected host platform (linux/amd64/v3) and no specific platform was requested
exec /usr/local/bin/docker-entrypoint.sh: exec format error
```

So my question is : why does this work ?

Reply:
I'm fine with merging this if we remove everything related to arm64 and postpone new archs to a different PR.
I believe we'd have to remove this changes to this file, change the platforms from the yaml file and maybe remove the qemu step (but I'd rather not touch this file, since I have no idea how to test all of this)

Reply:
Hello, I can verify that the arm64 built is working as expected. I actually created the fork for that, the thing is that you are right, it should not work but it does 🤣. I believe that it is related to Qemu, the Github actions runner is amd64 after all and maybe Qemu perform the necessary steps to fix the problems to cross compile the image.

Reply:
Hey anthares101, thanks for your work. Your feedback on this is greatly appreciated.

It makes absolutely no sense, yet it works.

Assuming you have static qemu binaries installed and binfmt configured properly, you can reproduce the build with (this is slow):
```sh
docker buildx build --platform linux/arm64 front
```

I am not sure if I should be impressed that there is some logic to find a similar version of a pinned container for an other architecture or scared that Docker ignores digests whenever it feels like.